.travis.yml

```diff
@@ -1,27 +1,91 @@
 language: go
 
-matrix:
-  include:
-  -  os: linux
-     sudo: required
-     services:
-       - docker
-  -  os: osx
-
 go:
   - 1.15.x
 
 notifications:
   email: false
 
-install:
-  # Ideally, the (brew update) should not be necessary and Travis would have fairly
-  # frequenstly updated OS images; that’s not been the case historically.
-  # In particular, explicitly unlink python@2, which has been removed from Homebrew
-  # since the last OS image build (as of July 2020), but the Travis OS still
-  # contains it, and it prevents updating of Python 3.
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update && brew unlink python@2 && brew install gpgme ; fi
-
-script:
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]];   then hack/travis_osx.sh ; fi
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then make vendor && ./hack/tree_status.sh && make local-cross && make check ; fi
+env:
+  global:
+    - REGISTRY=quay.io
+
+# Just declaration of the image-build-push step with script actions to execute
+x_base_steps:
+  - &image-build-push
+    services:
+      - docker
+    os: linux
+    script:
+      # skopeo upstream image build and push
+      - export SOURCE_TYPE=upstream; export REPO=skopeo/upstream; export TAG=master
+      - make build-arch-container
+      # Dont's push if it's PR check
+      - if [[ "$TRAVIS_EVENT_TYPE" != "pull_request" ]]; then make push-arch-container ; fi
+      # skopeo stable image build and push
+      - export SOURCE_TYPE=stable; export REPO=skopeo/stable; export TAG=v1.2.0; export EXTRA_REPO=containers/skopeo
+      - make build-arch-container
+      # Dont's push if it's PR check
+      - if [[ "$TRAVIS_EVENT_TYPE" != "pull_request" ]]; then make push-arch-container ; fi
+
+# Just declaration of stage order to run
+stages:
+  # Test for local build
+  - local-build
+  # Build and push image for 1 architecture
+  - name: image-build-push
+    if: branch = master
+  # Create and push image manifest to have multi-arch image on top of arch specific images
+  - name: multi-arch-manifest-push
+    if: type != pull_request AND branch = master
+
+# Actual execution of steps
+jobs:
+  allow_failures:
+  - if: arch in (s390x, ppc64le)
```

Comment:
I'm not sure I understand why failures should be ignored here, shouldn't we want this to fail if something breaks on those arches?

Reply:
just from my experience, many communities don't want to block their x86 builds/releases if build fails for other architectures. So it's more for community comfort. Anyway, the error is visible and not hidden in Travis, so it's always an option to go and fix it.
Also based on my experience, the non-amd64 errors are quite rare. 
So it's up to reviewers, definitely I can remove this statement.

Reply:
Ahh okay, that smells like sound reasoning.  Ya, I too am fine leaving it up to others to weigh in on.

Reply:
How would ignoring failures actually work with the multi-arch images? If we require x86 to succeed and ignore a ppc failure, would we still run the multi-arch push and now end up with a multi-arch image that contains a fresh x86 version and an obsolete ppc version (from some previous successful ppc run)? That seems very undesirable.

Reply:
ok, I will remove this questionable check from the pipeline